evap/static/ts/src/utils.ts

```diff
@@ -13,10 +13,36 @@ export function assertDefined<T>(val: T): asserts val is NonNullable<T> {
     assert(val !== null);
 }
 
+export function assertDefinedUnwrap<T>(val: T): NonNullable<T> {
+    assertDefined(val);
+    return val;
+}
+
 export const sleep = (ms?: number): Promise<number> => {
     return new Promise(resolve => window.setTimeout(resolve, ms));
 };
 
+// based on: https://docs.djangoproject.com/en/3.1/ref/csrf/#ajax
+export function getCookie(name: string): string | null {
+    if (document.cookie !== "") {
+        const cookie = document.cookie
+            .split(";")
+            .map(cookie => cookie.trim())
+            .find(cookie => cookie.substring(0, name.length + 1) === `${name}=`);
+        if (cookie) {
+            return decodeURIComponent(cookie.substring(name.length + 1));
+        }
+    }
+    return null;
+}
+
+export function setCookie(key: string, value: string): void {
+    document.cookie = key + "=" + value + ";path=/;SameSite=strict";
+}
+
+export const testable = {
+    getCookie,
+};
```

Comment:
@niklasmohrin what is the purpose of this indirection? Do we need it?

Reply:
we could probably get rid of the `testable` object everywhere, don't really know why it was introduced, `export function` should work just fine. however, I think that it's best to just stick to the existing convention for this PR